File: 11기 코딩테스트_230308/연습2/MostAppearInteger/MostAppearInteger.js
/**
 * @param n {number}
 * @returns {number}
 */
function solution(n) {
  const numCounter = []

  while (1 <= n) {
    const i = Math.floor(n % 10)
    numCounter[i] = numCounter[i] + 1 || 1

    n /= 10
  }

  let mostAppearNumber = undefined
  let appearCount = 0
  numCounter.forEach((count, num) => {
    if (appearCount < count) {
      mostAppearNumber = num
      appearCount = count
    }
  })

  return mostAppearNumber
}

export default solution

File: 11기 코딩테스트_230308/연습2/MostAppearInteger/MostAppearInteger.py
import math


def solution(n):
    """
    :param n: number
    :return: number
    """

    num_counter = {}
    while (1 <= n):
        i = math.floor(n % 10)
        num_counter[i] = num_counter[i] + 1 if i in num_counter else 1

        n /= 10

    most_appear_number = None
    appear_count = 0
    for num, count in sorted(num_counter.items()):
        if appear_count < count:
            most_appear_number = num
            appear_count = count

    return most_appear_number
